### DISCUSSION TITLE
想学习graphscope源码，应该从哪开始

### DISCUSSION THREAD

**Discussion by nanyoullm** [alibaba/GraphScope#1513] [General]
想学习graphscope源码，应该从哪开始

**Comment by yecol** [comment]
Hi @nanyoullm 

Thanks for your interest in GraphScope!
Unfortunately, we are still working on documentation for guiding how to read the code. 

Here let me try to give a brief intro to the main components of GraphScope.

<img width="990" alt="image" src="https://user-images.githubusercontent.com/426463/167241487-938f185d-0c39-43e9-baf6-382dd102b831.png">

1) Python Interface, the python client for end-users;
2) coordinator: it manages the k8s cluster and receives all requests and triages them to the responding engine;
3) analytical_engine: for iterative graph algorithms, which extends GRAPE;
4) interactive_engine: for gremlin queries, please note that the fold is being deprecated and upgraded to GAIA;
5) learning_engine: for GNN, it was built on graph-learn;
6) vineyard: an in-memory data manager, graphs on all engines are managed by vineyard;
7) groot: a persistent store for graph.

Since the codebase is still in change, I would suggest the learning path can be:
1) -> 2) -> 3) -> 6) -> 4) -> 5) -> 7)

The components and their code path are shown in the figure. 
Hope you enjoy reading the code!
If you have any further questions, please do not hesitate to contact us. :)

**Reply by nanyoullm** [reply]
非常感谢提供指导！

**Reply by nanyoullm** [reply]
学习之路艰难，卡在了第一步😢
https://github.com/alibaba/GraphScope/discussions/1598

**Reply by YijianLiu** [reply]
> 学习之路艰难，卡在了第一步😢 #1598

@nanyoullm 可以一起交流呀

**Reply by yecol** [reply]
@YijianLiu 也可以加入我们的钉钉群 随时找我们~

**Comment by JackyYangPassion** [comment]
钉钉群有邀请码吗

**Reply by sighingnow** [reply]
钉钉群号：31533139
